core/src/main/java/org/wildfly/extras/creaper/core/online/HttpModelControllerClient.java

```diff
@@ -157,21 +157,53 @@ public OperationResponse executeOperation(Operation operation, OperationMessageH
     public void close() throws IOException {
     }
 
+    protected ModelNode parseResponse(CloseableHttpResponse response) throws IOException {
+        ModelNode result;
+        String content = EntityUtils.toString(response.getEntity());
+        int status = response.getStatusLine().getStatusCode();
+        if (status == HttpStatus.SC_OK) {
+            result = ModelNode.fromJSONString(content);
+        } else {
+            result = new ModelNode();
+            result.get(Constants.OUTCOME).set(Constants.FAILED);
+            result.get(Constants.FAILURE_DESCRIPTION).set(content);
```

Comment:
Well that's a question. I honestly don't know.

I'm leaning towards throwing an exception. Returning a `ModelNode` with the `outcome` of `failed` means that the server did execute the operation, found that there's something wrong, rollbacked it, and sent a result. If there would be any other problem, the native client would throw an exception, and so should we.